src/vm/vm_core.rs

```diff
@@ -4015,4 +4134,90 @@ mod tests {
         let expected_traceback = vec![(Relocatable::from((1, 2)), Relocatable::from((0, 34)))];
         assert_eq!(vm.get_traceback_entries(), expected_traceback);
     }
+
+    #[test]
+    fn builder_test() {
+        let virtual_machine_from_builder: VirtualMachine = VirtualMachineBuilder::default()
+            .run_finished(true)
+            .current_step(12)
+            ._program_base(Some(MaybeRelocatable::from((1, 0))))
+            .accessed_addresses(Some(vec![Relocatable::from((1, 0))]))
+            .builtin_runners(vec![(
+                "string",
+                BuiltinRunner::from(HashBuiltinRunner::new(12, true)),
+            )])
+            .run_context(RunContext {
+                pc: Relocatable::from((0, 0)),
+                ap: 18,
+                fp: 0,
+            })
+            .segments(MemorySegmentManager {
+                segment_sizes: HashMap::new(),
+                segment_used_sizes: Some(vec![1]),
+                public_memory_offsets: HashMap::new(),
+                memory: Memory::new(),
+            })
+            .skip_instruction_execution(true)
+            .trace(Some(vec![TraceEntry {
+                pc: Relocatable::from((0, 1)),
+                ap: Relocatable::from((0, 1)),
+                fp: Relocatable::from((0, 1)),
+            }]))
+            .build();
```

Comment:
About the test of the hook.

Don't call `.build()` immediately. Assign the builder to a variable instead.

Then behind `#[cfg(feature = "hooks")]` call `hooks` on this variable.

Then call `build()` and store the vm_from_builder as you did.

Then do all your assert, then behind `#[cfg(feature = "hooks")]` add one more assert for the hooks

Reply:
Smart !